File: app/src/main/java/com/stupidtree/cloudliter/ui/welcome/signup/SignUpTrigger.java
package com.stupidtree.cloudliter.ui.welcome.signup;

import com.stupidtree.cloudliter.data.model.UserLocal;
import com.stupidtree.cloudliter.ui.base.Trigger;

public class SignUpTrigger extends Trigger {

    private String username;
    private String password;
    private UserLocal.GENDER gender;
    private String nickname;

    public static SignUpTrigger getRequestState(String username, String password,
                                                UserLocal.GENDER gender, String nickname
                                              ){
        SignUpTrigger ls =  new SignUpTrigger();
        ls.username = username;
        ls.password = password;
        ls.gender = gender;
        ls.nickname = nickname;
        ls.setActioning();
        return ls;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    public String getGender() {
        return gender== UserLocal.GENDER.MALE?"MALE":"FEMALE";
    }

    public String getNickname() {
        return nickname;
    }
}

File: app/src/main/java/com/stupidtree/cloudliter/ui/welcome/signup/SignUpTrigger.kt
package com.stupidtree.cloudliter.ui.welcome.signup

import com.stupidtree.cloudliter.data.model.UserLocal.GENDER
import com.stupidtree.component.data.Trigger

class SignUpTrigger : Trigger() {
    var username: String? = null
        private set
    var password: String? = null
        private set
    private var gender: GENDER? = null
    var nickname: String? = null
        private set

    fun getGender(): String {
        return if (gender === GENDER.MALE) "MALE" else "FEMALE"
    }

    companion object {
        @JvmStatic
        fun getRequestState(username: String?, password: String?,
                            gender: GENDER?, nickname: String?
        ): SignUpTrigger {
            val ls = SignUpTrigger()
            ls.username = username
            ls.password = password
            ls.gender = gender
            ls.nickname = nickname
            ls.setActioning()
            return ls
        }
    }
}